IPIP/0000-gateway-redirects.md

```diff
@@ -0,0 +1,76 @@
+# IPIP 0000: Gateway Redirects
+
+- Start Date: (format: 2022-06-15)
+- Related Issues:
+  - [ipfs/specs/issues/257](https://github.com/ipfs/specs/issues/257)
+  - [ipfs/kubo/pull/8890](https://github.com/ipfs/kubo/pull/8890)
+
+## Summary
+
+Provide support for URL redirects and rewrites for web sites hosted on Subdomain or DNSLink Gateways, thus enabling support for [single-page applications (SPAs)](https://en.wikipedia.org/wiki/Single-page_application), and avoiding  [link rot](https://en.wikipedia.org/wiki/Link_rot) when moving to IPFS-backed hosting.
+
+## Motivation
+
+Web sites often need to redirect from one URL to another, for example, to change the appearance of a URL, to change where content is located without breaking existing links (see [Cool URIs don't change](https://www.w3.org/Provider/Style/URI), [link rot](https://en.wikipedia.org/wiki/Link_rot)), to redirect invalid URLs to a pretty 404 page, or to enable URL rewriting.
+URL rewriting in particular is a critical feature for hosting SPAs, allowing routing logic to be handled by front end code. SPA support is the primary impetus for this RFC.
+
+Currently the only way to handle URL redirects or rewrites is with additional software such as NGINX sitting in front of the Gateway. This software introduces operational complexity and decreases the uniformity of experience when navigating to content hosted on a Gateway, thus decreasing the value proposition of hosting web sites in IPFS.
+
+This IPIP proposes the introduction of redirect support for content hosted on Subdomain or DNSLink Gateways, configured via a `_redirects` file residing underneath the root CID of the web site.
+
+## Detailed design
+
+Allow developers to configure redirect support by adding redirect rules to a file named `_redirects` stored underneath the root CID of their web site.
+The format for this file is similar to those of [Netlify](https://docs.netlify.com/routing/redirects/#syntax-for-the-redirects-file) and [Cloudflare Pages](https://developers.cloudflare.com/pages/platform/redirects) but only supporting a subset of their functionality.
```

Comment:
Is it a strict subset? Deviation would not be nice here

Reply:
Currently it's a strict subset, but my thinking is that IPFS will have different tradeoffs to consider that will warrant not attempting to stay fully in sync.